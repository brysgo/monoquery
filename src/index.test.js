import { createMonoQuery, gql } from "./";
import graphql from "graphql-tag";
import { print } from "graphql/language";

test("build, run distribute simple query", async () => {
  const fragments = {
    simpleFragment: gql`
      fragment SimpleFragment on Query {
        something
      }
    `,
    withGraphQLTag: graphql`
      fragment AnotherFragment on Query {
        anotherThing
      }
    `
  };
  const monoQuery = createMonoQuery(({ query, variables, operationName }) => {
    expect(print(query)).toMatchSnapshot();
    return Promise.resolve({
      data: {
        hello: "world",
        something: "good",
        anotherThing: "hey"
      }
    });
  });
  const result = await monoQuery({
    query: gql`
      {
        hello
        ...SimpleFragment
        ...AnotherFragment
      }
      ${fragments.simpleFragment}
      ${fragments.withGraphQLTag}
    `
  });
  expect(result.getResultsFor(fragments)).toEqual({
    simpleFragment: { something: "good" },
    withGraphQLTag: { anotherThing: "hey" }
  });
});
