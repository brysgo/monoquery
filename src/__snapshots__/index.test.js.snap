// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build, run distribute simple query 1`] = `
"{
  hello
  ...SimpleFragment
  ...AnotherFragment
}

fragment SimpleFragment on Query {
  something
}

fragment AnotherFragment on Query {
  anotherThing
}
"
`;
